docs/proposals/scheduling/20220701-schedule-gang.md

```diff
@@ -0,0 +1,421 @@
+
+### Non Goals and Future Work
+1. Provide ability to solve Gang resource deadlock problems with `non-strict-mode`.
+
+## Proposal
+
+### Key concept
+
+#### strict-mode and non-strict-mode
+
+As mentioned above, in `strict-mode`, if a pod failed to be scheduled, the pods that have been successfully scheduled in 
+this scheduling cycle will be rolled back, and the remaining pods waiting for scheduling will be rejected in 
+PreFilter check util this scheduling cycle passed. We call this mode is `strict-mode`.
+
+In `non-strict-mode`, if a pod failed to be scheduled, it has no impact on any other pod. We will continue to accumulate 
+the allocated pod until the condition of Gang is met. This process is friendly to Gangs with large number of pods, but it 
+will increase the risk of resource deadlock between Gangs. For example, the quota of the quota group is 10(quota will be proposed later), 
+and the user submits three Gangs with 5 pods. Due to various plugin constraints, Gang1\2\3 may allocate resources of 3\3\4 respectively. 
+Since the quota group's quota is full, there will be no new resource scheduling. We call this is resource deadlock of resource Gang.
+In future proposal, we will try to fix this problem.
+
+#### GangGroup
+As mentioned above, Some jobs have complex Gang requirements. For example, a job has several roles. Each role will have several pods 
+and its own Gang conditions. Jobs also need different roles to form different GangGroups. All pods in a GangGroup can 
+trigger the bind process only after all roles in a GangGroup meet their Gang conditions. So we introduce `GangGroup` concept,
+which allow user to bundle different Gangs together.
+
+#### After Gang
+It should be noted that, if the resource accumulation conditions of Gang are met, then some pods failed in the process of binding,
+or some bound pods are preempted\rescheduled, should the constraints of Gang still be effective in the process of resource reallocation? 
+Because the initial purpose of Gang is to require pods to be pulled up at the same time, if some pods have been pulled up, 
+then the subsequent Gang behavior is meaningless. Therefore, when once Gang has been satisfied, all subsequent resource allocations 
+are no longer constrained by Gang rules, and their performance is similar to ordinary pod.
+
+As mentioned above, `WaitTime` is the max wait time since first pod comes to permit stage. If `WaitTime` is timeout, 
+scheduler will roll back all assumed pods, update each pod's annotation with `gang.scheduling.koordinator.sh/timeout=true`, and
+won't schedule these pods anymore. User should pay attention to this status and delete pods timely.
+
+### API
+#### Definition
+
+Our original intention is to improve and enhance the ability of the community's original `PodGroup`, so we will be 
+compatible with the way the community declares the `PodGroup`. We also provide a lighting way to just use annotations to 
+use Gang feature.
+
+#### Annotation way
```

Comment:
Move the annotation way under the PodGroup? This will make it easier for users to understand why this mode is supported. In addition, when the annotation mode is used, you can ignore the expression of the compatible community (because the community has migrated to PodGroup).

Reply:
fixed